components/kyma-environment-broker/internal/hyperscaler/azure/client.go

```diff
@@ -42,11 +66,55 @@ func (nc *AzureClient) CreateNamespace(ctx context.Context, azureCfg *Config, gr
 	// we need to use a copy of the location, because the following azure call will modify it
 	locationCopy := azureCfg.GetLocation()
 	parameters := eventhub.EHNamespace{Location: &locationCopy, Tags: tags}
-	ehNamespace, err := nc.createAndWait(ctx, groupName, namespace, parameters)
+	ehNamespace, err := nc.createAndWaitNamespace(ctx, groupName, namespace, parameters)
 	return &ehNamespace, err
 }
 
-func (nc *AzureClient) createAndWait(ctx context.Context, resourceGroupName string, namespaceName string, parameters eventhub.EHNamespace) (result eventhub.EHNamespace, err error) {
+func (nc *AzureClient) DeleteResourceGroup(ctx context.Context, tags Tags) (resources.GroupsDeleteFuture, error) {
+	// get name of resource group
+	resourceGroup, err := nc.GetResourceGroup(ctx, tags)
+	if err != nil {
+		return resources.GroupsDeleteFuture{}, err
+	}
+
+	// trigger async deletion of the resource group
+	if resourceGroup.Name == nil {
+		return resources.GroupsDeleteFuture{}, fmt.Errorf("resource group name is nil")
+	}
+	nc.logger.Infof("deleting resource group: %s", *resourceGroup.Name)
+	future, err := nc.resourcegroupClient.Delete(ctx, *resourceGroup.Name)
+	if err != nil {
+		return resources.GroupsDeleteFuture{}, err
+	}
+	return future, err
+}
+
+func (nc *AzureClient) GetResourceGroup(ctx context.Context, tags Tags) (resources.Group, error) {
+	if tags[TagInstanceID] == nil {
+		return resources.Group{}, fmt.Errorf("serviceInstance is nil")
+	}
+
+	serviceInstanceID := *tags[TagInstanceID]
+	filter := fmt.Sprintf("tagName eq 'InstanceID' and tagValue eq '%s'", serviceInstanceID)
+
+	// we only expect one ResourceGroup, so not using pagination here should be fine
+	resourceGroupIterator, err := nc.resourcegroupClient.ListComplete(ctx, filter, nil)
+	if err != nil {
+		return resources.Group{}, err
+	}
+	for resourceGroupIterator.NotDone() {
+		resourceGroup := resourceGroupIterator.Value()
+
+		// there should only be one resource group with the given service instance id
```

Comment:
@nachtmaar AFAICU, if the iterator returned more than one value, the last value will win.
IMHO, we should panic if there are more than one value. Assuming that if there is a bug we might delete a resource group of a different runtime. What do you think?

Reply:
Lines 109-112 should be prevent that by returning an error. Did you observe a different behaviour?

Reply:
Fixed